chapter4/src/4_2_1/main.rs

```diff
@@ -0,0 +1,19 @@
+use tokio;
+
+#[tokio::main]
+async fn main() {
+    println!("start sub()");
+
+    // 終了を受け取るためのchannel
+    let (done_tx, mut done_rx) = tokio::sync::mpsc::channel(1);
+
+    tokio::spawn(async move {
+        println!("sub() is finished");
+        done_tx.send(true);
```

Comment:
```suggestion
        done_tx.send(true).await.unwrap();
```

が良いと思います。
元のコードだと、SendError が発生したときにメインスレッドが `done_rx.recv().await` で永遠にブロックする挙動になりそうです。